sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/channel/DefaultChannel.java

```diff
@@ -339,93 +369,103 @@ private synchronized void triggerIngestion(final @NonNull String groupName) {
 
         /* Get a batch from Persistence. */
         final List<Log> batch = new ArrayList<>(groupState.mMaxLogsPerBatch);
+        final int currentState = mStateChangeCounter;
```

Comment:
Can you rename the local variable? Getting confused about  **State**, better to have **Counter**